parm/data_locations.yml

```diff
@@ -236,6 +236,20 @@ RAP:
     file_names:
       <<: *rap_file_names
 
+RRFS:
+  hpss:
+    protocol: htar
```

Comment:
This section doesn't seem to have any hpss information in it. I also don't know that HPSS houses any of the RRFS runs. Please collapse this down into only the `aws` section.

Reply:
It does not, but the structure was still appeared necessary because of other internal code dependency. SRW sets naming conventions for the on-premises files from their names set for hpss.  When this is removed no data naming is otherwise specified for local files on a disk, and model would proceed on pulling the data from AWS even if it exists locally.

Reply:
It is not. You need to define the file names in the first section that is defined.

Reply:
A test on Hera fails fails to locate files on disk when modified version of data_locations.yaml is set with the following entries for RRFS:

```
RRFS:
  aws:
    protocol: download
    url: https://noaa-rrfs-pds.s3.amazonaws.com/rrfs_a/rrfs_a.{yyyymmdd}/{hh}/control/
    file_names: &rrfs_file_names
      anl:
        - rrfs.t{hh}z.prslev.f{fcst_hr:03d}.conus.grib2
      fcst:
        - rrfs.t{hh}z.prslev.f{fcst_hr:03d}.conus.grib2
```

Please see the log files in 
`/scratch1/NCEPDEV/nems/Natalie.Perlin/SRW/expt_dirs/test_no_HPSS_set/log/`, 
`get_extrn_ics_2024060517.log`, `get_extrn_lbcs_2024060517.log` 

```
This is the ex-script for the task that copies or fetches external model
input data from disk, HPSS, or a URL, and stages them to the
workflow-specified location so that they may be used to generate initial
or lateral boundary conditions for the FV3.
========================================================================
INFO: Logging level set to DEBUG

Running script retrieve_data.py with args:
--------------------------------------------------------------------------------
--------------------------------------------------------------------------------
       file_set: anl
     cycle_date: 2024-06-05 17:00:00
    data_stores: ['disk', 'disk']
      data_type: RRFS
       fcst_hrs: [0]
    output_path: /scratch1/NCEPDEV/nems/Natalie.Perlin/SRW/expt_dirs/test_no_HPSS_set/2024060517/RRFS/for_ICS
    ics_or_lbcs: ICS
        version: None
        symlink: True
          debug: True
 file_templates: None
       file_fmt: None
input_file_path: /scratch1/NCEPDEV/nems/role.epic/UFS_SRW_data/develop/input_model_data/RRFS/2024060517
        members: None
   summary_file: extrn_mdl_var_defns.sh
     check_file: False
--------------------------------------------------------------------------------
--------------------------------------------------------------------------------
INFO: Checking disk for RRFS

Traceback (most recent call last):
  File "/scratch1/NCEPDEV/nems/Natalie.Perlin/SRW/srw-dev-rrfs/ush/retrieve_data.py", line 1081, in <module>
    main(sys.argv[1:])
  File "/scratch1/NCEPDEV/nems/Natalie.Perlin/SRW/srw-dev-rrfs/ush/retrieve_data.py", line 835, in main
      file_templates = get_file_templates(
                     ^^^^^^^^^^^^^^^^^^^
  File "/scratch1/NCEPDEV/nems/Natalie.Perlin/SRW/srw-dev-rrfs/ush/retrieve_data.py", line 330, in get_file_templates
    raise argparse.ArgumentTypeError(msg)
argparse.ArgumentTypeError: No file naming convention found. They must be provided                 either on the command line or on in a config file.
FATAL ERROR:
ERROR:
  From script:  "JREGIONAL_GET_EXTRN_MDL_FILES"
  Full path to script:  "/scratch1/NCEPDEV/nems/Natalie.Perlin/SRW/srw-dev-rrfs/jobs/JREGIONAL_GET_EXTRN_MDL_FILES"
Call to ex-script corresponding to J-job "JREGIONAL_GET_EXTRN_MDL_FILES" failed.
Exiting with nonzero status.
_______________________________________________________________
Start Epilog on node hfe08 for job 62421979 :: Tue Jun 25 13:53:25 UTC 2024
Job 62421979 finished for user Natalie.Perlin in partition service with exit code 1:0
_______________________________________________________________
End Epilogue Tue Jun 25 13:53:25 UTC 2024
```

The following location in ./ush/retrieve_data.py (lines 834-840) sets the naming templates for the files on disk to that set for hpss:

```
        if data_store == "disk":
            file_templates = get_file_templates(
                cla,
                known_data_info,
                data_store="hpss",
                use_cla_tmpl=True,
            )
```